src/main/java/com/commercetools/sync/customers/helpers/CustomerBatchValidator.java

```diff
@@ -0,0 +1,223 @@
+package com.commercetools.sync.customers.helpers;
+
+import com.commercetools.sync.commons.helpers.BaseBatchValidator;
+import com.commercetools.sync.commons.helpers.BaseSyncStatistics;
+import com.commercetools.sync.customers.CustomerSyncOptions;
+import io.sphere.sdk.customers.CustomerDraft;
+import io.sphere.sdk.models.Address;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class CustomerBatchValidator
+    extends BaseBatchValidator<CustomerDraft, CustomerSyncOptions, BaseSyncStatistics> {
```

Comment:
Or we should use `CustomerSyncStatistics` here? since we don't expect to parse `SyncStatistics` class of other resources as argument.

Reply:
Good catch, I was developing that before the CustomerSyncStatistic object is added, and then forgot to change it.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/587/commits/721edc11b784fe4db9efaa484405ce15b0115bf5